test/libyul/StackShufflingTest.cpp

```diff
@@ -14,41 +14,172 @@
+        );
 
-BOOST_AUTO_TEST_CASE(swap_cycle)
-{
-	std::vector<Scope::Variable> scopeVariables;
-	Scope::Function function;
-	std::vector<VariableSlot> v;
-	for (size_t i = 0; i < 17; ++i)
-		scopeVariables.emplace_back(Scope::Variable{""_yulstring, YulString{"v" + to_string(i)}});
-	for (size_t i = 0; i < 17; ++i)
-		v.emplace_back(VariableSlot{scopeVariables[i]});
-
-	Stack sourceStack{
-		v[0], v[1], v[2], v[3], v[4], v[5], v[6], v[7], v[9], v[10], v[11], v[12], v[13], v[14], v[15], v[16],
-		FunctionReturnLabelSlot{function}, FunctionReturnLabelSlot{function}, v[5]};
-	Stack targetStack{
-		v[1], v[0], v[2], v[3], v[4], v[5], v[6], v[7], v[9], v[10], v[11], v[12], v[13], v[14], v[15], v[16],
-		FunctionReturnLabelSlot{function}, JunkSlot{}, JunkSlot{}
+		scanner->next();
+		string functionName = scanner->currentLiteral();
+		vector<Expression> arguments{};
+		while (scanner->next() != Token::RBrack)
+		{
+			auto token = scanner->currentToken();
+			if (token == Token::Comma)
+				continue;
+			arguments.emplace_back(Expression{Identifier{{}, YulString(scanner->currentLiteral())}});
+		}
+		soltestAssert(
+			scanner->currentToken() == Token::RBrack,
+			"Invalid stack layout syntax: expected \"]\"."
+        );
+		return yul::FunctionCall{
+			{},
+			yul::Identifier{{}, YulString(functionName)},
+			arguments,
+	    };
+	};
+
+	auto parseStack = [&](Scanner& scanner, Stack& stack) -> bool
+	{
+		soltestAssert(scanner.currentToken() == Token::LBrack, "Invalid stack layout syntax: expected \"[\".");
+		scanner.next();
+		while (scanner.currentToken() != Token::RBrack &&
+			   scanner.currentToken() != Token::EOS)
+		{
+			string literal = scanner.currentLiteral();
+			if (literal == "RET")
+			{
+				scanner.next(); // read [
+				if (scanner.currentToken() == Token::LBrack)
+				{
+					auto call = readFunctionCall(&scanner);
+					stack.emplace_back(FunctionCallReturnLabelSlot{
+							m_functions.insert(
+								make_pair(call.functionName.name, call)
+							).first->second
+					});
+				}
+				else
+				{
+					static Scope::Function function;
+					stack.emplace_back(FunctionReturnLabelSlot{function});
+					continue;
+				}
+			}
+			else if (literal == "TMP")
+			{
+				scanner.next(); // read [
+				auto call = readFunctionCall(&scanner);
+				stack.emplace_back(TemporarySlot{
+						m_functions.insert(make_pair(call.functionName.name, call)).first->second
+				});
+			}
+			else if (literal.find("0x") != string::npos || scanner.currentToken() == Token::Number)
+			{
+				stack.emplace_back(LiteralSlot{u256(literal)});
+			}
+			else if (literal == "JUNK")
+			{
+				stack.emplace_back(JunkSlot());
+			}
+			else if (literal == "GHOST")
+			{
+				scanner.next(); // read [
+				scanner.next(); // read number of ghost variables as ghostVariableId
+				string ghostVariableId = scanner.currentLiteral();
+				Scope::Variable ghostVar = Scope::Variable{""_yulstring, YulString(literal + "_" + ghostVariableId)};
```

Comment:
```suggestion
				Scope::Variable ghostVar = Scope::Variable{""_yulstring, YulString(literal + "[" + ghostVariableId + "]")};
```
better to avoid any potential conflict with other variable names.